shumai/tensor/stats.ts

```diff
@@ -22,12 +22,86 @@ export function collectStats(tensors: Tensor[]) {
     }
     for (const key of Object.keys(t.stats)) {
       if (key in stats) {
+        stats[key].count += t.stats[key].count
         stats[key].time += t.stats[key].time
         stats[key].bytes += t.stats[key].bytes
+        stats[key].ops += t.stats[key].ops
       } else {
         stats[key] = t.stats[key]
       }
     }
   }
   return stats
 }
+
+export type StatsEntry = {
+  count: number
+  time: number
+  bytes: bigint
+  ops: bigint
+}
+
+export type StatsSummary = {
+  totals: StatsEntry
+  perSec: StatsEntry
+  entriesByStack: Map<string, StatsEntry>
+  entriesByOp: Map<string, StatsEntry>
+}
+
+export function getStatsSummary(tensors: Tensor[], priorStats?: StatsSummary): StatsSummary {
+  let stats: StatsSummary = priorStats
+  if (!stats) {
+    // init
+    stats = {
+      totals: { count: 0, time: 0, bytes: 0n, ops: 0n },
+      perSec: { count: 0, time: 0, bytes: 0n, ops: 0n },
+      entriesByStack: new Map(),
+      entriesByOp: new Map()
+    }
+  }
+
+  for (const t of tensors) {
+    if (!t.stats) {
+      throw new Error(`tensor has no stats ${t.ptr}:${t.shape}`)
+    }
+
+    for (const key of Object.keys(t.stats)) {
+      const op = /^(\n)?(.*)@/.exec(key)?.[2] || key
```

Comment:
why not `t.op`? perhaps I don't understand this line

Reply:
Tensors track stats across tensor operations so we must pluck the actual operation from the stack trace.